pandas/core/ops/__init__.py

```diff
@@ -325,8 +405,9 @@ def dispatch_to_series(left, right, func, str_rep=None, axis=None):
     elif isinstance(right, ABCDataFrame):
         assert right._indexed_same(left)
 
-        def column_op(a, b):
-            return {i: func(a.iloc[:, i], b.iloc[:, i]) for i in range(len(a.columns))}
```

Comment:
An alternative would be to improve the performance of this line, and avoid the complexity that is added in `operate_blockwise`. 
It will not give the same performance boost as what you now have with the blockwise operation, but it might already be a nice speedup with much less complex code, so a trade-off to consider IMO.

Reply:
regardless this is worth doing on L343/424 and L348/429